### PR TITLE
Allow the Dockerfile to be self-built and add documentation on how to…

### LINKED ISSUE #27: docker webmvc-example
Currently, when I do performance troubleshooting, I start frontend and backend locally and use `wrk -t4 -c128 -d100s http://localhost:8081 --latency` against our docker-compose setup which includes grafana. I then look at the results.

Some of our stuff uses cloud and also docker internal networks are a bit weird. It seems like we could create and openzipkin/example-frontend and openzipkin/example-backend image which could then be deployed and tweaked as necessary. These literally could be added to our default docker-compose file for people who don't already have apps also.. or at least an additional compose file could add them.

I'm big in favor of literally using this repo vs something less used, relevant or tested. Sleuth report more concerns than many people do, and we have a lot of branches here which means we can switch to working and relevant config somewhat quickly. I want to re-use energy we already spend here (referencing this repo several times a week) vs creating something similar that people don't use today.

cc @openzipkin/core

### PR DESCRIPTION
**Pull request by anuraaga**
… run it.

Open to any suggestions. I went with a single image with commands to control backend / frontend since the difference of separating out with a `-`, or a `:`, or a ` ` didn't seem that big, but this ensures the same code is only downloaded once onto a dev's machine without other trickiness. I think it also provides the flexibility e.g., for kafka, mentioned on the issue.

Now that the `Dockerfile` builds by itself, I suggest we set up Docker Hub to automatically build on master push - it's a simple and good way of publishing docker images for at least this sort that doesn't use Maven Central as a source of truth.

I am thinking after this is published, we could add it to the normal `openzipkin/docker` `docker-compose.yml`.

Fixes #27

### CODE REVIEW

**Comment by anuraaga** [comment]
Ah - now I know what the docker here was for :) Will take a look at the acceptance tests

**Comment by anuraaga** [comment]
Finally ready for review - somehow the parts that changed significantly, frontend / backend, were easy to get working in the acceptance tests but I couldn't get the fetch of the zipkin query, which previously wasn't executed in the container, to work, despite not having touched the zipkin server. Finally switched to running the fetch in the container seems to have solved it but not sure why this changr was necessary.

With this approach, the continuous build ends up building the code twice, once to run normal tests in circle ci, and once to run acceptance tests in docker. While we'd usually want to optimize this, given this builds very fast and isn't updated so often, I'm leaning towards letting it slide.

**Review by marcingrzejszczak — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -122,6 +122,26 @@ Under the covers, this uses the brave-opentracing bridge:
...
+A docker image containing the example can be built using
+
+```bash
+$ docker build -t openzipkin/example-sleuth-webmvc -f docker/Dockerfile .
+```
+
+The backend can be started with a command similar to
+
+```bash
+$ docker run -it --rm -p 9000 openzipkin/example-sleuth-webmvc backend
```
Comment (codefromthecrypt) [inline README.md]: like the arg and the prefix of example makes sense

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,21 @@
-FROM circleci/openjdk:8-jdk
+FROM maven:3-jdk-11-slim
```
Comment (codefromthecrypt) [inline docker/Dockerfile:1]: slim like a million megs I presume? :D

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,6 +1,21 @@
...
-COPY . /code
-WORKDIR /code
+ADD . /code
 
-ENTRYPOINT ["./docker/run.sh"]
+RUN cd /code && mvn -B --no-transfer-progress package -DskipTests=true
+
+FROM gcr.io/distroless/java:11-debug
+
+RUN ["/busybox/sh", "-c", "adduser -g '' -h /zipkin -D zipkin"]
```
Comment (codefromthecrypt) [inline docker/Dockerfile:9]: not in love with the username and work dir as it looks more like what's used in zipkin, and this isn't zipkin. however, I'm lacking another option and don't really care :P

File: docker/run.sh [hunk 4]
```diff
@@ -1,6 +1,19 @@
...
+    docker run -it --rm -p 8081 openzipkin/example-sleuth-webmvc frontend
+    docker run -it --rm -p 9000 openzipkin/example-sleuth-webmvc backend"
+    exit 1
+esac
 
-CLASS_NAME="$1"; shift
-java -cp "target/dependency/*:target/*" -Dlogging.level.org.springframework.cloud.sleuth=DEBUG -Dspring.zipkin.baseUrl=http://zipkin:9411 -Dspring.example.backendBaseUrl=http://backend:9000 sleuth.webmvc.$CLASS_NAME
+java -cp target/sleuth-webmvc-*-exec.jar -Dlogging.level.org.springframework.cloud.sleuth=DEBUG \
+  -Dspring.zipkin.baseUrl=http://zipkin:9411 -Dspring.example.backendBaseUrl=http://backend:9000 \
+  -Dloader.main=sleuth.webmvc.$CLASS_NAME org.springframework.boot.loader.PropertiesLauncher
```
Comment (codefromthecrypt) [inline docker/run.sh:19]: slick

**Review by codefromthecrypt — APPROVED** [review]
thanks for this!

**Pull request merged**